File: android/worker/app/src/main/java/com/workforcex/worker/api/JobBrowseItem.java
package com.workforcex.worker.api;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;

public class JobBrowseItem implements Serializable {
    @SerializedName("id")
    public String id;
    @SerializedName("title")
    public String title;
    @SerializedName("companyName")
    public String companyName;
    @SerializedName("skillsRequired")
    public String skillsRequired;
    @SerializedName("experienceRequired")
    public Integer experienceRequired;
    @SerializedName("location")
    public String location;
    @SerializedName("salaryMin")
    public Double salaryMin;
    @SerializedName("salaryMax")
    public Double salaryMax;
    @SerializedName("openPositions")
    public Integer openPositions;
    @SerializedName("description")
    public String description;
    public Boolean applied = false; // local UI state — not from API
}

File: android/worker/shared/src/commonMain/kotlin/com/workforcex/shared/models/JobBrowseItem.kt
package com.workforcex.shared.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class JobBrowseItem(
    @SerializedName("id")
    val id: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("companyName")
    val companyName: String,
    @SerializedName("skillsRequired")
    val skillsRequired: String,
    @SerializedName("experienceRequired")
    val experienceRequired: Int,
    @SerializedName("location")
    val location: String,
    @SerializedName("salaryMin")
    val salaryMin: Double,
    @SerializedName("salaryMax")
    val salaryMax: Double,
    @SerializedName("openPositions")
    val openPositions: Int,
    @SerializedName("description")
    val description: String,
    var applied: Boolean = false // local UI state — not from API
) : Serializable